matter_server/server/ota/provider.py

```diff
@@ -0,0 +1,366 @@
+
+            # And write. This is persistent, so only need to be done after we commissioned
+            # the OTA Provider App.
+            write_result: Attribute.AttributeWriteResult = (
+                await chip_device_controller.write_attribute(
+                    ota_provider_node_id,
+                    [(0, Clusters.AccessControl.Attributes.Acl(acl_list))],
+                )
+            )
+            if write_result[0].Status != Status.Success:
+                logging.error(
+                    "Failed writing adjusted OTA Provider App ACL: Status %s.",
+                    str(write_result[0].Status),
+                )
+                raise UpdateError("Error while setting up OTA Provider.")
+        except ChipStackError as ex:
+            logging.exception("Failed setting up OTA Provider.", exc_info=ex)
+            raise UpdateError("Error while setting up OTA Provider.") from ex
+
+    async def start_update(
+        self, chip_device_controller: ChipDeviceControllerWrapper, node_id: int
+    ) -> None:
+        """Start the OTA Provider and trigger the update."""
+
+        self._ota_target_node_id = node_id
+
+        loop = asyncio.get_running_loop()
+
+        ota_provider_passcode = secrets.randbelow(2**21)
+        ota_provider_discriminator = secrets.randbelow(2**12)
+
+        timestamp = datetime.now(tz=UTC).strftime("%Y%m%d_%H%M%S")
+        ota_provider_cmd = [
+            "chip-ota-provider-app",
+            "--passcode",
+            str(ota_provider_passcode),
+            "--discriminator",
+            str(ota_provider_discriminator),
+            "--secured-device-port",
+            "0",
+            "--KVS",
+            str(self._ota_provider_dir / f"chip_kvs_ota_provider_{timestamp}"),
+            "--filepath",
+            str(self._ota_file_path),
+        ]
+
+        log_file_path = self._ota_provider_dir / f"ota_provider_{timestamp}.log"
+
+        log_file = await loop.run_in_executor(None, log_file_path.open, "w")
+
+        try:
+            LOGGER.info("Starting OTA Provider")
+            self._ota_provider_proc = await asyncio.create_subprocess_exec(
+                *ota_provider_cmd, stdout=log_file, stderr=log_file
+            )
+
+            self._ota_provider_task = loop.create_task(
+                self._ota_provider_proc.communicate()
+            )
+
+            # Commission and prepare ephemeral OTA Provider
+            LOGGER.info("Commission and initialize OTA Provider")
+            ota_provider_node_id = (
+                DEFAULT_OTA_PROVIDER_NODE_ID + self._ota_target_node_id
+            )
+            await self._commission_ota_provider(
+                chip_device_controller,
+                ota_provider_passcode,
+                ota_provider_discriminator,
+                ota_provider_node_id,
+            )
+
+            # Notify update node about the availability of the OTA Provider. It will query
+            # the OTA provider and start the update.
```

Comment:
```suggestion
            # Notify update node about the availability of the OTA Provider. 
            # It will query the OTA provider and start the update.
```